### DISCUSSION TITLE
Serverless Functions per Month limitation, who hits it ?

### DISCUSSION THREAD

**Discussion by fabien-h** [vercel/vercel#4927] [General]
I need to assess Vercel for a client project and I dug a little bit on the limitations and this one surprises me a little bit :

`Serverless Functions per Month: 160 | 640`

We can deploy 3000 times per day. But we can only edit our deployed functions 640 times per month globaly. If I change an utility and it updates 20 functions in 5 projects, next time I push to github, it will use 100 credits. So I've just used 15% of my monthly credit. It seems super low to me. Do I understand right ?

If yes, we cannot have an API that can be updated for 640 edits per month.

**Comment by samcx** [comment]
You are correct → https://vercel.com/docs/v2/platform/limits#serverless-functions-per-month-(hobby). Serverless function creation is much more costly, so this is unfortunately the case.

Can you further elaborate on your use-case? Maybe we can consider you for an Enterprise plan with custom limits, or if possible consider refactors to better fit the limits.

**Comment by fabien-h** [comment]
> Serverless function creation is much more costly, so this is unfortunately the case.

That sounds super weird to me. There must be something I misunderstand about your architecture. I've worked with the serverless framework (https://www.serverless.com/) for a while and took the habit of having many small specific endpoints. Since creating a lambda is free on AWS and I pay only for the execution cost. For pricing optimization, I was inclined to split all my functions into the various subcases instead of having a forest of "if" at the beginning of the handler. And everything runs in a few ms on a cheap 128 Mo lambda. it also makes the code of each endpoint super super short, easy to debug and to maintain, since it makes so little, it's easy to be sure that you are not breaking something else.

The use case is not super exotic. In the pro plan, I can make 640 edits per month to my functions. And it seems not very much.

> if possible consider refactors to better fit the limits.

Absolutely. Functions are functions. I can have a single endpoint `/api/v1` with an URL parameter like `/api/v1?endpoint=createUser` that will define the actual endpoint and a big fat switch at the beginning of the handler. It means that I can deploy my entire API 640 times per month including as many edits per endpoints that I want. And that seems enough.

If you tell me that it's good for you, it's good for me. Since you deploy on AWS, it just "feels" like the execution cost will be higher. But since I'm paying a flat 20$ price to you and no execution cost, that's not on me.

**Comment by sztadii** [comment]
I am thinking if the "Vercel" will go to the "pey_per_request" policy it will be game-changing in web development. I found the tool yesterday and I am really surprised that it is so easy to use.
